azure-mgmt-resources/src/main/java/com/azure/management/BearerTokenAuthenticationPolicy.java

```diff
@@ -72,7 +72,8 @@ private String getDefaultScopeFromRequest(HttpRequest request) {
 
     @Override
     public Mono<HttpResponse> process(HttpPipelineCallContext context, HttpPipelineNextPolicy next) {
-        if ("http".equals(context.getHttpRequest().getUrl().getProtocol().toLowerCase())) {
+        if ("http".equals(context.getHttpRequest().getUrl().getProtocol().toLowerCase()) &&
+            !"localhost".equals(context.getHttpRequest().getUrl().getHost().toLowerCase().split(":")[0])) {
```

Comment:
Try see if we can have another solution? It seems not reasonable to change product code because of test reason.

Reply:
I use it due to I thought the `localhost` should not be blocked even there is user use it in real request. And the request for `localhost` usually doesn't have `https`.

Reply:
Yes, probably OK. Just check whether we could say use an overloaded BearerTokenPolicy for playback (or do we need Auth on playback at all?).

Reply:
I think we could add a `skipAuthetication` for `restClientBuilder`.